docs/architecture/app-management.md

```diff
@@ -0,0 +1,307 @@
+    name: String!
+    description: String
+    labels: Labels
+    webhooks: [WebhookInput!]
+    healthCheckURL: String
+    apis: [APIDefinitionInput!]
+    eventAPIs: [EventAPIDefinitionInput!]
+    documents: [DocumentInput!]
+    integrationSystemName: String #optional
+    labels: [LabelInput!]
+}
+```
+IntegrationSystemID is an optional. It means that you can still create Application that is not managed by IntegrationSystem.
+
+### Example
+In this example, Integration System is registered and then it configures newly added Application. 
+1. Register Integration System
+```graphql
+mutation {
+    createIntegrationSystem(in: {name: "simpleIntegrationSystem"} ) {
+      id
+      name
+    }
+}
+
+```
+2. Create Application with specified `integrationSystemName`
+```graphql
+mutation {
+    createApplication(in:{name:"simpleApplication", integrationSystemName:"simpleIntegrationSystem"}) {
+        id
+        name
+        integrationSystemName
+        labels
+    }
+}
+```
+Compass adds label with name `integrationSystemName` for just created Application, so output of the previous mutation is the following:
+```
+{
+  "data": {
+    "createApplication": {
+      "id": "d046590f-934f-411f-91e2-d446b404a2a2",
+      "name": "simpleApplication",
+      "integrationSystemName": "simpleIntegrationSystem",
+      "labels": {
+        "scenarios":["DEFULT"],
+        "integrationSystemName":"simpleIntegrationSystem",
+      },
+      
+    }
+  }
+}
+```
+
+Thanks to that,  Integration System can easily fetch all dependant Applications by querying them by label.
+Then, Integration System is responsible for updating Application details, like registering API and events.
+
+## Integration System supporting many Application types
+In the previous example, there was an assumption that every Application managed by given Integration System represents the same type
+of the Application.  
+In case IntegrationSystem supports many types of Applications, information about Application type should be stored in the Application labels.
+Let assume that IntegrationSystem supports two types of Applications: `ecommerce` and `marketing` and such information 
+is stored in label `simpleIntegrationSystem/applicationType`.
+To create an Application of type `ecommerce`, use following mutation:
+```graphql
+mutation {
+    createApplication(in:{name:"ecommerceApp", integrationSystemName:"simpleIntegrationSystem", labels:[{key:"simpleIntegrationSystem/applicationType",value:"ecommerce"}]}) {
+        id
+        name
+        integrationSystemName
+        labels
+    }
+}
+```
+Drawback of this approach is that label name (`simpleIntegrationSystem/applicationType`) and supported values(`ecommerce`,`marketing`) are arbitrary 
+defined by a IntegrationSystem and has to be documented. 
+Luckily, IntegrationSystem can simplify this process by defining ApplicationTemplates to explicitly specify supported types.
+
+## Managing ApplicationTemplates
```

Comment:
I'm missing here a bit of background regarding the placeholders mechanism. Do we support only simple key-value pairs, or some function calls? Do we support some optional placeholders? Do we validate during the creation that all placeholders are filled with appropriate values?